tests/vm_delete_protection_test.go

```diff
@@ -0,0 +1,87 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/rand"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"kubevirt.io/ssp-operator/tests/env"
+)
+
+const deleteProtectionLabel = "kubevirt.io/vm-delete-protection"
+
+var _ = Describe("VM delete protection", func() {
+
+	var vm *kubevirtv1.VirtualMachine
+
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	AfterEach(func() {
+		err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
+		if !errors.IsNotFound(err) {
+			Eventually(func(g Gomega) error {
+				err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
+				if err != nil {
+					return err
+				}
+
+				vm.Labels[deleteProtectionLabel] = "False"
+				return apiClient.Update(ctx, vm)
+			}, env.ShortTimeout(), time.Second).Should(Succeed())
+
+			Expect(apiClient.Delete(ctx, vm)).To(Succeed())
+			waitForDeletion(client.ObjectKeyFromObject(vm), &kubevirtv1.VirtualMachine{})
+		}
```

Comment:
Can you simply Delete the VM, without calling Get and Update?
```
if vm != nil {
	Expect(apiClient.Delete(ctx, vm)).To(Or(Succeed(), MatchError(errors.IsNotFound, "errors.IsNotFound")))
	vm = nil
}

```

Reply:
That would be risky, IMHO. If for some reason, a VM stays with the protection and policy enabled, the test suite won't be able to delete the VM. In this way, we are sure that the protection for these particular VMs is removed.

Reply:
Right, of course. Ignore my previous comment, I didn't realize that.